docs/SplunkOperatorUpgrade.md

```diff
@@ -0,0 +1,73 @@
+# How to upgrade Splunk Operator and Splunk Enterprise Deployments
+
+To upgrade Splunk Operator Kubernetes(SKO), you must install new version directly on top of old version(On same Kubernetes Cluster/Namespace). When Splunk Operator Uprades it updates CRDs and Splunk PODs deployed by the Operator to match the latest SPEC. An upgrade of the Splunk Operator will initiate a restart of all deployed pods. If the Splunk Enterprise Docker image changes, the deployed pods will be upgraded and restarted. 
+In a typical Splunk Multisite Indexer Cluster with Search Head Cluster the upgrade will follow the below order:
+* A new Splunk Operator pod will be created and the existing operator pod will drop
+* If a License Master is present. License Master pod will be terminated and new pod with updated Spec will be created.
+* After new License Master pod comes up Cluster Master and Deployer pods will be terminated
+* New Cluster Master and Deployer pod are crated with updated spec
+* After Cluster Master Pod is ready, Indexer Pods are reset one by one per site. 
+* After Deployer Pod is ready, Search Heads are reset one by one.
+* Once all the members of Indexer Cluster and Search Head Cluster are ready, Monitoring Console Pod will be reset.
+
+## Prequisites 
+
+Before you upgrade, review the Splunk Operator [change log](https://github.com/splunk/splunk-operator/releases) page for information on changes made in the latest release.
+
+The Splunk Operator does not provide a means for downgrading to a previous version. If you need to revert to a prior Splunk Operator release, remove the current release and reinstall the version you want.
```

Comment:
"If you need to revert to a prior Splunk Operator release, remove the current release and reinstall the version you want."
We should explain the implications of doing this